pyuvdata/uvdata/mir_parser.py

```diff
@@ -10,1082 +10,6383 @@
+                for hid in int_dict:
+                    int_dict[hid]["chan_avg"] *= chanavg_dict[
+                        getattr(self, attr).get_value("iband", header_key=hid)
+                    ]
+
+            chan_avg_arr = [chanavg_dict[band] for band in getattr(self, attr)["iband"]]
+            if attr == "sp_data":
+                self._rechunk_data(self.vis_data, chan_avg_arr, inplace=True)
+                self._rechunk_raw(self.raw_data, chan_avg_arr, inplace=True)
+            else:
+                self._rechunk_data(self.auto_data, chan_avg_arr, inplace=True)
+
+    def __add__(self, other, merge=None, overwrite=None, force=False, inplace=False):
+        """
+        Add two MirParser objects.
+
+        This method allows for combining MirParser objects under two different
+        scenarios. In the first, which we call a "merge", two objects are instantiated
+        from the same file, but may have different data loaded due to, for example,
+        different calls to `select` being run. In the second scenarion, which we call
+        a "concatenation", objects are instantiated from different files, which need
+        to be recombined (e.g., a single track is broken in half due to an
+        intervening observation/system hiccup/etc.).
+
+        Note that while some checking is performed to check if the metadata objects
+        look identical, no checking is done on the `vis_data`, `raw_data`, and
+        `auto_data` attributes (other than that they exist). If either object does not
+        have data loaded, then the resultant object will also have no data loaded.
+
+        Parameters
+        ----------
+        other : MirParser object
+            Other MirParser object to combine with this data set.
+        overwrite : bool
+            If set to True, metadata from `other` will overwrite that present in
+            this object, even if they differ. Default is False.
+        merge : bool
+            If set to True, assume that the objects originate from the amd file, and
+            combine them together accordingly. If set to False, assume the two objects
+            originate from _different_ files, and concatenate them together. By default,
+            the method will check the internal file dictionary to see if it appears the
+            two objects come from the same file(s), automatically choosing between
+            merging or concatenating.
+        force : bool
+            If set to True, bypass certain checks to force the method to combine the
+            two objects. Note that this option should be used with care, as it can
+            result in objects with duplicate data (which may affect downstream
+            processing), or loss of support for handling auto-correlations within
+            the object. Default is False.
+        inplace : bool
+            If set to True, replace this object with the one resulting from the
+            addition operat
```

Comment:
Could use a comment, I had to look up that it meant "absolute tolerance", and maybe also comment why it needs to be different than the default in np.where.